### ISSUE TITLE
Nexjts SSR broken functionality with reactStrictMode enabled

### ISSUE THREAD

**Issue by abusada** [algolia/react-instantsearch#3537]
## 🐛 Bug description

I'm following [this guide](https://www.algolia.com/doc/guides/building-search-ui/going-further/server-side-rendering/react-hooks/?client=index.tsx#with-nextjs) trying to achieve server side rendering with nextjs and react-instantsearch-hooks.

once i wrap my code with `InstantSearchSSRProvider`, search functionality simple stops working,
![screenshot- 2022-06-23 at 15 57 31](https://user-images.githubusercontent.com/2572320/175317280-8d5260bc-4011-41ab-be58-e5e7e6c296d0.gif)

**Live reproduction:**

https://codesandbox.io/s/hungry-stitch-b070o2?file=/pages/index.tsx:990-1014

## 🔍 Bug reproduction

**Steps to reproduce the behavior:**

1. Go to the provided sandbox: https://codesandbox.io/s/hungry-stitch-b070o2?file=/pages/index.tsx:990-1014
2. Try using the search box

## 💭 Expected behavior

Expect search to work and hits to update, notice how in the [withour-ssr page](https://b070o2.sse.codesandbox.io/without-ssr) the search functionality is working properly.

![screenshot- 2022-06-23 at 16 03 39](https://user-images.githubusercontent.com/2572320/175318566-9a3e3c7c-a257-49f9-ac6b-1a482490f35b.gif)

## Environment

- React InstantSearch Hooks version: [6.29.0]
- React version: [18.2.0]
- Nextjs version [12.1.6]

please **note** the following configs in `next.config.js`:

```
module.exports = {
  reactStrictMode: true,
  swcMinify: true
};
```

**update**
setting `reactStrictMode` to false fixes the issue, here is a fork of the above sandbox with strict mode turned off: https://codesandbox.io/s/lucid-cache-n7h0no?file=/next.config.js:19-45

**Comment by abusada** [comment]
Could you possibly point me to the code where you suspect the issue is happening? perhaps i could be of help!

**Comment by Haroenv** [comment]
There's some tests:

- https://github.com/algolia/react-instantsearch/pull/3518/files#diff-7accb274a0dfc2e4d20af5c09f1c9b6f800efaedcc568ffa1fd06dc5f8de55eeR160
- https://github.com/algolia/react-instantsearch/pull/3518/files#diff-e64eff9c1a1f08e5af58847734df1f2a89ae5ec7cae42f9c0d644c0b8c64be6fR382
- https://github.com/algolia/react-instantsearch/blob/89e5db61081fd1581c7ac98a5b788ebfed85c9ff/packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts#L67 -> this should be called only on a timeout, as in useWidget

Also there's a more complex part where widgets get added more than once in hydration, but that might not be the cause of the issue

**Comment by francoischalifour** [comment]
This issue may be addressed in [v6.30.2](https://github.com/algolia/react-instantsearch/releases/tag/v6.30.2). Could you please upgrade and let us know how it goes?

**Comment by FabienMotte** [comment]
@kanehara Did the fix mentioned by @francoischalifour fixed your issue?

We're doing a round of cleanup before migrating this repository to the new [InstantSearch monorepo](https://github.com/algolia/instantsearch.js). This issue seems not to have generated much activity lately, so we're going to close it, feel free to reopen if needed.
